custom-elements/reactions/DOMTokenList.html

```diff
@@ -0,0 +1,223 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Custom Elements: CEReactions on DOMTokenList interface</title>
+<meta name="author" title="Ryosuke Niwa" href="mailto:rniwa@webkit.org">
+<meta name="assert" content="add, remove, toggle, replace, and the stringifier of DOMTokenList interface must have CEReactions">
+<meta name="help" content="https://dom.spec.whatwg.org/#node">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="./resources/reactions.js"></script>
+</head>
+<body>
+<div id="log"></div>
+<script>
+
+test(function () {
+    var element = defineNewCustomElement(['class']);
+    var instance = document.createElement(element.name);
+    assert_array_equals(element.log().types(), ['constructed']);
+    instance.classList.add('foo');
+    var logEntries = element.log();
+    assert_array_equals(logEntries.types(), ['attributeChanged']);
+    assert_equals(logEntries.log().name, 'class');
+    assert_equals(logEntries.log().oldValue, null);
+    assert_equals(logEntries.log().newValue, 'foo');
+    assert_equals(logEntries.log().namespace, null);
+}, 'add on DOMTokenList must enqueue a attributeChanged reaction when adding an attribute');
+
+test(function () {
+    var element = defineNewCustomElement([]);
+    var instance = document.createElement(element.name);
+    assert_array_equals(element.log().types(), ['constructed']);
+    instance.classList.add('foo');
+    assert_array_equals(element.log().types(), []);
+}, 'add on DOMTokenList must not enqueue a attributeChanged reaction when adding an unobserved attribute');
+
+test(function () {
+    var element = defineNewCustomElement(['class']);
+    var instance = document.createElement(element.name);
+    instance.setAttribute('class', 'hello');
+    assert_array_equals(element.log().types(), ['constructed', 'attributeChanged']);
+    instance.classList.add('world');
+    var logEntries = element.log();
+    assert_array_equals(logEntries.types(), ['attributeChanged']);
+    assert_equals(logEntries.log().name, 'class');
+    assert_equals(logEntries.log().oldValue, 'hello');
+    assert_equals(logEntries.log().newValue, 'hello world');
+    assert_equals(logEntries.log().namespace, null);
+}, 'add on DOMTokenList must enqueue a attributeChanged reaction when adding a value to an existing attribute');
+
+test(function () {
+    var element = defineNewCustomElement([]);
+    var instance = document.createElement(element.name);
+    instance.setAttribute('class', 'hello');
+    assert_array_equals(element.log().types(), ['constructed']);
+    instance.classList.add('world');
+    assert_array_equals(element.log().types(), []);
+}, 'add on DOMTokenList must enqueue a attributeChanged reaction when adding a value to an existing unobserved attribute');
```

Comment:
It must *not* enqueue an attributeChanged reaction. The test looks right but the description is confusing.